### ISSUE TITLE
Focus pulled to first menu item when activating menu with keyboard

### ISSUE THREAD

**Issue by niamhmadden24** [Sage/carbon#7264]
### Description

When using keyboard to open the menu, focus is pulled to the first menu item. This fails WCAG 2.4.3 Focus Order. 

It's expected that focus stays on the menu itself once opened, until tab is pressed to go to the first menu item. 

### Reproduction

https://carbon.sage.com/?path=/docs/menu--docs

### Steps to reproduce

Use keyboard to open the menu

### Carbon version

152.2.0

### React version

v18

### Relevant browsers

Chrome

### Relevant OSs

MacOS

### Confidentiality

- [x] I confirm there is no confidential or commercially sensitive information included.

**Comment by niamhmadden24** [comment]
The same issue occurs on the multi-action button:
 https://carbon.sage.com/?path=/docs/multi-action-button--docs

**Comment by edleeks87** [comment]
FE-7154

**Comment by edleeks87** [comment]
We are going to need to review the behaviour in the `MultiActionButton` and `SplitButton` with accessibility as I believe that behaviour was added to support voiceover
